base/bucket.go

```diff
@@ -112,6 +120,49 @@ func (spec BucketSpec) IsWalrusBucket() bool {
 	return strings.Contains(spec.Server, "walrus:")
 }
 
+func (spec BucketSpec) IsTLS() bool {
+	return strings.HasPrefix(spec.Server, "couchbases") || strings.HasPrefix(spec.Server, "https")
+}
+
+func (spec BucketSpec) UseClientCert() bool {
+	if spec.Certpath == "" || spec.Keypath == "" {
+		return false
+	}
+	return true
+}
+
+// Builds a gocb connection string based on BucketSpec.Server.
+// Adds idle connection configuration, and X.509 auth settings when
+// certpath/keypath/cacertpath specified.
+func (spec BucketSpec) GetGoCBConnString() (string, error) {
+
+	connSpec, err := gocbconnstr.Parse(spec.Server)
+	if err != nil {
+		return "", err
+	}
+
+	// Increase the number of idle connections per-host to fix SG #3534
+	if connSpec.Options == nil {
+		connSpec.Options = map[string][]string{}
+	}
+	asValues := url.Values(connSpec.Options)
+	asValues.Set("http_max_idle_conns_per_host", DefaultHttpMaxIdleConnsPerHost)
+	asValues.Set("http_max_idle_conns", DefaultHttpMaxIdleConns)
+	asValues.Set("http_idle_conn_timeout", DefaultHttpIdleConnTimeoutMilliseconds)
+
+	if spec.Certpath != "" && spec.Keypath != "" {
+		asValues.Set("certpath", spec.Certpath)
```

Comment:
It might be worth checking if the path actually exists on the filesystem at this point and aborting early if not.  (could make it slightly easier to debug).

Reply:
If this was needed I'd recommend we do it during config validation, but the error gets bubbled up clearly as-is:

```
2018-05-14T16:41:50.994-07:00 [INF] Configured process to allow 5000 open file descriptors
2018-05-14T16:41:50.994-07:00 [INF] Opening db /x509 as bucket "x509Bucket", pool "default", server <couchbases://127.0.0.1>
2018-05-14T16:41:50.994-07:00 [INF] GoCBCustomSGTranscoder Opening Couchbase database x509Bucket on <couchbases://127.0.0.1> as user "x509Bucket"
2018-05-14T16:41:50.998-07:00 [INF] Auth: gocb connect returned error: open /Userd/adam/Desktop/x509/ca.pem: no such file or directory
```

Reply:
Ok, great